resources/metadata/abapAddonAssemblyKitCheckCVs.yaml

```diff
@@ -0,0 +1,68 @@
+metadata:
+    name: abapAddonAssemblyKitCheckCVs
+    description: This step checks the validity of Software Component Versions.
+    longDescription: |
+      This steps takes a list of Software Component Versions from the addonDescriptorFileName and checks whether they exist or are a valid successor of an existing Software Component Version.
+      It resolves the dotted version string into version, support package level and patch level and writes it to the commonPipelineEnvironment.
+
+spec:
+  inputs:
+    secrets:
+      - name: abapAddonAssemblyKitCredentialsId
+        description: Credential stored in Jenkins for Addon Assembly Kit as a Service System (AAKaaS)
+        type: jenkins
+    params:
+      - name: abapAddonAssemblyKitEndpoint
+        type: string
+        description: Base URL to the Addon Assembly Kit as a Service System (AAKaaS)
+        scope:
+        - PARAMETERS
+        - STAGES
+        - STEPS
+        - GENERAL
+        mandatory: true
+      - name: username
+        type: string
+        description: User for the Addon Assembly Kit as a Service System (AAKaaS)
+        scope:
+        - PARAMETERS
+        - STAGES
+        - STEPS
+        mandatory: true
+        secret: true
+      - name: password
+        type: string
+        description: Password the Addon Assembly Kit as a Service System (AAKaaS)
+        scope:
+        - PARAMETERS
+        - STAGES
+        - STEPS
+        mandatory: true
+        secret: true
+      - name: addonDescriptorFileName
+        type: string
+        description: File name of the YAML file which describes the Product Version and corresponding Software Component Versions
+        mandatory: true
+        default: addon.yml
+        scope:
+          - PARAMETERS
+          - STAGES
+          - STEPS
+          - GENERAL
+      - name: addonDescriptor
+        type: string
+        description: Structure in the commonPipelineEnvironment containing information about the Product Version and corresponding Software Component Versions
+        mandatory: false
+        scope:
+          - PARAMETERS
+          - STAGES
+          - STEPS
+        resourceRef:
+          - name: commonPipelineEnvironment
+            param: abap/addonDescriptor
+  outputs:
+    resources:
+      - name: commonPipelineEnvironment
+        type: piperEnvironment
+        params:
+          - name: abap/addonDescriptor
```

Comment:
Can `commonPipelineEnvironment` handle this parameter? Do we need to write this into the `custom` section or create a dedicated `abap` section?

Reply:
The `commonPipelineEnvironment` can handle `abapRepositoryNames`: https://github.com/SAP/jenkins-library/blob/master/vars/commonPipelineEnvironment.groovy#L191

But I think we should use [this](https://github.com/SAP/jenkins-library/blob/master/vars/commonPipelineEnvironment.groovy#L50-L58) more generic method for storing further parameters. It can also handle [json outputs](https://github.com/SAP/jenkins-library/blob/master/vars/commonPipelineEnvironment.groovy#L219).

Reply:
I assume there were changes in commonPipelineEnvironment.groovy, which were not included in this PR?

Reply:
I do not see any in the original PR #1973 and in the hand over I was told this was discussed and decided to go that way...

Reply:
As there are some obsolete parts in commonPipelineEnvironment.groovy, I adapted the current coding so that it will work for this PR:
https://github.com/SAP/jenkins-library/pull/2035

abapRepositoryNames is superseded by abapAddonDescriptor.